### ISSUE TITLE
Linear Mode Connectivity under Data Shifts for Deep Ensembles of Image Classifiers

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#477]
## Links
https://arxiv.org/abs/2511.04514

### 🔹3行要約
本論文は、データシフト（分布変化）がニューラルネットの**線形モード接続（Linear Mode Connectivity: LMC）**に与える影響を体系的に解析した研究である。  
学習率とバッチサイズの設定がSGDノイズを制御し、異なる局所解間の接続性と汎化性能を左右することを実験的に示した。  
LMCにより得られるモデルは似た誤りを繰り返す傾向があり、アンサンブルの多様性と効率性のトレードオフを明らかにしている。

<img width="756" height="977" alt="Image" src="https://github.com/user-attachments/assets/4b182b58-63af-47ad-9d68-7cc3b36eb151" />

---

### 1. 背景と目的
線形モード接続（LMC）は、複数の学習済みモデル間が「低損失の線形経路」で結ばれる現象を指す。  
これにより、異なる初期値やデータサブセットで学習されたモデル間の汎化特性を理解でき、**アンサンブル学習・分散最適化・モデル統合**などに応用できる。  
本研究は、**データシフト（covariate shift・label imbalance・domain shift）**がLMCをどのように破壊または維持するかを実験的に解析する。

---

### 2. 研究の新規性と既存研究との差異
従来のLMC研究（Frankle et al., 2020; Entezari et al., 2022）は、主に**同一データ分布下の初期値の違い**や**SGDノイズ**の影響を扱っていた。  
本研究はこれを拡張し、**実際の分布変動（データシフト）**をノイズ源として捉え、  
「学習率」「バッチサイズ」がノイズスケールを通じて学習の安定性に与える影響を定量的に検証した点が新しい。  
また、**ResNet/VGG/MLP**といった異構造モデル間の挙動比較を行い、深さや正規化層の有無によるLMCの出現条件を包括的に評価している。

---

### 3. 方法論（Methodology）
- **データセット**：MNIST, CIFAR-10, BigEarthNet  
  - ラベル不均衡やバンド差異によりデータシフトを人工的に導入。
- **モデル構成**：MLP, VGG, ResNet（バッチ正規化あり/なし）  
- **評価指標**：
  - *Loss Barrier*：線形補間経路上の損失最大値と端点の平均損失の差  
  - *Accuracy Difference (Δ)*：補間モデルの精度変化  
  - *Cosine Similarity・Manhattan距離*：モデル間のパラメータ類似度  
  - *Wrong Agreement / Disagreement*：誤分類の一致率・不一致率（アンサンブル多様性の指標）

---

#### (1) Covariate Shiftの影響（Fig.4）
小バッチ学習では、同一SGDノイズ下でもデータシフトにより異なる局所解へ収束する。  
浅いMLPは安定している一方、ResNetや深いVGGはバリアが顕著で、性能が最大15%低下した。

#### (2) Overfittingの影響（Fig.5–6）
線形補間曲線が滑らかでも真のLMCとは限らず、**過学習**により損失が低く見える偽の安定状態が発生する。  
Batch Normalization（BN）を導入すると過学習が軽減し、真のLMCを回復。

#### (3) 大バッチ・低学習率の効果（Fig.7）
バッチサイズを増加、または学習率を低減させることで**SGDノイズスケール**が減少し、  
モデルが同一局所解に収束しやすくなり、LMCが「再現」される。  
ただし、ResNetなど深いモデルでは依然として不安定性が残る。

#### (4) ラベル不均衡とドメインシフト（Fig.8–9）
ResNetはラベル不均衡に対して最も不安定で、特徴の再利用構造（skip connection）が影響している。  
また、BigEarthNetでのバンド差異が大きい場合、バリアが小さくても**汎化性能が悪化**する。

#### (5) 入力正規化とBNの相互作用（Fig.10–12）
入力画像を[0,1]正規化するとLMCが崩壊しやすくなるが、BNの導入で回復。  
ReLU活性化と重み分布の符号不均衡が情報損失を生むためと解釈される。

#### (6) モデル類似度とノイズスケール（Fig.13–15）
大バッチ・小学習率ではモデル間の角度（cos類似度）が小さくなり、パラメータ空間で近づく。  
ただし、MLPでは「ほぼ直交」していてもLMCを示す場合があり、構造依存の効果が見られる。

#### (7) アンサンブル性能（Table III, Fig.17）
LMCで得たモデル群は同じ誤りを繰り返す傾向があり、  
異なる初期化で学習したモデル群よりもアンサンブル性能が低い（ResNetで約6%低下）。  
→ **LMCは訓練効率を高めるが、多様性を損なう**というトレードオフを明示。

---

### 5. 主要な図の説明
- **Fig.1**：SGDノイズスケール g = ε(N/B − 1) の模式図。大バッチ・低学習率でノイズが減り、安定性向上。  
- **Fig.4–7**：データシフトやバッチサイズ・学習率の違いがLoss Barrierに与える影響を可視化。  
- **Fig.13–15**：モデル間の角度と距離の関係、深層モデルほど不安定な学習動態を示すことを極座標で表示。  
- **Fig.17**：LMC由来モデルと異初期化モデルのアンサンブル比較。LMCモデルは誤りの一致率が高い。

---

### 6. 貢献の実証方法
- **複数データセット・モデル構造**における包括的な実験設計。  
- **ノイズ源の固定実験**により、データシフト由来の効果を純粋に分離。  
- **定量指標（Barrier, Δ, WA, WD）**によるLMCの定義と評価。  
- **実験再現性検証**（Appendix B–E）を通じ、SGDノイズの制御と再現性を確認。

---

### 7. 結論と意義
- LMCの成立は、**SGDノイズスケール**・**学習率/バッチサイズ**・**データ分布**の相互作用で決まる。  
- LMCは訓練安定性と効率を高めるが、**アンサンブル多様性の低下**という限界を持つ。  
- 今後は、LMCの角度範囲と損失地形（flat/sharp minima）の幾何学的関係を理論的に結び付ける研究が期待される。

---

### 📘 まとめ
本研究は、LMCを「SGDノイズとデータシフトの関数」として再定義し、  
深層モデルの安定性・汎化・多様性のトレードオフ構造を明確に示した点で重要である。  
実験的に、LMCは「過学習耐性」と「訓練効率」をもたらす一方で、  
モデル間誤差の相関を高めるという**深層アンサンブルの根本的限界**を実証している。
